OpenRA.Mods.Common/Traits/World/Locomotor.cs

```diff
@@ -57,7 +57,11 @@ public static class CustomMovementLayerType
 
 	[TraitLocation(SystemActors.World | SystemActors.EditorWorld)]
 	[Desc("Used by Mobile. Attach these to the world actor. You can have multiple variants by adding @suffixes.")]
-	public class LocomotorInfo : TraitInfo
+	public class LocomotorInfo : TraitInfo,
+		After<TerrainTunnelLayerInfo>,
+		After<SubterraneanActorLayerInfo>,
+		After<JumpjetActorLayerInfo>,
+		After<ElevatedBridgeLayerInfo>
 	{
```

Comment:
Could we make it work on Interface, so we could write `After<ICustomMomentLayer>` and it would load after all the concrete types ?

Reply:
`After<>` supports that, because it's borrowing the same logic from `Requires<>` which already supports that.

The only reason I didn't do that here is we don't actually have a `ICustomMomentLayerInfo` to rely on. Might be nice to generalise the custom movement layers (in another PR) so we can support adding extra ones rather than having the hardcoded list.

Reply:
I believe that there are downstream mods that implement their own custom layer types, so IMO we should implement that interface (not exposing any methods - just a tag) as part of this PR instead of hardcoding the types.

Reply:
I was overthinking for a wider generalization, an interface is sensible, added.